### ISSUE TITLE
GDPR / privacy concerns

### ISSUE THREAD

**Issue by matthijskooijman** [evolution-events/Artaxerxes#43]
Some privacy/GDPR/AVG-related observations.

 - Most data we store will be personal data, while medical data will be special personal data. Food allergies seems to also count as medical data (as implied by e.g. http://de-bovenkamer.nl/privacybeleid/ or https://www.ngsmassage.nl/download/820 or https://www.splend.nl/gdpr-privacy-event-marketing/)
 - Storing personal data is only allowed based on a ground specified in [article 6](https://gdpr-info.eu/art-6-gdpr/). For special personal data, additionally one of the grounds in [article 9](https://gdpr-info.eu/art-9-gdpr/) is needed.
 - Data we store related to an event subscription, would be related to a contract we have (or intend to have) with the participant, so is lawful under 6.1.b.
 - Storing an e-mail address serves to send people mailings, and can be lawful under 6.1.a (explicit consent) or 6.1.f (because it serves the legitimate interests of the receiver). When someone unsubscribes from all mailing lists, it would still be relevant to store the e-mail address to allow account recovery.
 - Storing other data between events (e.g. address data) would be lawful under 6.1.a (consent), since we already plan to make that opt-in and easy to opt-out. This is less likely to be lawful under 6.1.f since the interests of "easier to register at future events" might not weigh up to the privacy risk of storing this data.
 - Storing medical data would be lawful under 6.1a and 9.2.a (both consent). 9.2.a is only allowed for "specified purposes", so we must be explicit about what we use this data for. Also, 6.1a talks about consent, while 9.2a talks about "explicit consent", which is slightly more strict (entering optional data when it is clear what the purpose is, would be consent, but not explicit consent). See also ["What is explicit consent?"](https://ico.org.uk/for-organisations/guide-to-the-general-data-protection-regulation-gdpr/consent/what-is-valid-consent/#what5). It seems this article does not specify any particular constraints on *how long* to store this data, but it is likely that "for a future event" does not count as "specified".
 - [Article 7](https://gdpr-info.eu/art-7-gdpr/) gives rules for consent. These are that consent must be proven afterwards (7.1), must be worded clearly and sufficiently separated from other content (7.2), can always be revoked (7.3, "It shall be as easy to withdraw as to give consent.") and that consent is not considered freely given if it is imposed as a condition for a contract (7.4).
 - [Article 12](https://gdpr-info.eu/art-12-gdpr/) provides rules about how to communicate and handle requests, as further specified in the subsequent articles.
 - [Article 13](https://gdpr-info.eu/art-13-gdpr/) & [article 14](https://gdpr-info.eu/art-14-gdpr/) state the obligation to inform a data subject when collecting personal data (directly when obtaining data from the subject, or up to 1 month later when observing data about the subject). When processing happens for other goals than previously specified, data subjects must be informed about this. For us, perhaps we should keep a privacy policy and track what version of it has been accepted / viewed by the user?
   Article 13.2d requires informing about the possibility to lodge a complaint with a supervisory authority, so we might need to figure out which authority this is (probably college bescherming persoonsgegevens).
 - [Article 15](https://gdpr-info.eu/art-15-gdpr/) specifies the right to request your own data, along with some accompanying info.
 - [Article 16](https://gdpr-info.eu/art-16-gdpr/) specifies the right to rectify your own data (without going into much detail on how that should work).
 - [Article 17](https://gdpr-info.eu/art-17-gdpr/) specifies the right to be forgotten, which is possible on a number of grounds. Most likely ground would be that data is "no longer necessary in relation to the purposes for which they were collected or otherwise processed" (17.1a). We can probably argue that at least keeping a user object with a name is necessary for the purpose of history-keeping, but all other data (including e-mailaddress, username, etc.) should probably be removed. For data where the only ground is consent (6.1a), this consent can always be removed and the data must be deleted (even if it would still be necessary! 17.1b). For data whose only ground is 6.1f (legitimate interests), there is an explicit right to object ([article 21](https://gdpr-info.eu/art-21-gdpr/) and 17.1.c). There are some other grounds, which likely won't apply to us.
 - [Article 18](https://gdpr-info.eu/art-18-gdpr/) specifies the right to restrict further processing. This would prevent further processing without removing the data, in cases where the data might be incorrect, unlawful, has a complaint pending, etc.
 - [Article 19](https://gdpr-info.eu/art-19-gdpr/) specifies the requirement to forward some requests to third parties that have been given access to the data.
 - [Article 20](https://gdpr-info.eu/art-20-gdpr/) specifies the right to data portability, meaning personal data must be made available on request in a portable electronic format, to be imported by another controller. This only applies to data processed with consent or in relation to a contract. For us, portability does not really apply, and we can figure this out if we ever get such a request.
 - [Article 21](https://gdpr-info.eu/art-21-gdpr/) specifies the right to object to processing when data is collected for "legitimate purposes" (6.1f) or public interest (6.1e), or when data is used for direct marketing. This might apply to us, in which case we must at least point out this possibility to our users.
 - [Article 22](https://gdpr-info.eu/art-22-gdpr/) specifies limitations on processing of personal data for automated decision making.
 - [Article 23](https://gdpr-info.eu/art-23-gdpr/) provides rules for other laws that want to restrict people's rights.
 - [Article 24](https://gdpr-info.eu/art-24-gdpr/), [Article 25](https://gdpr-info.eu/art-25-gdpr/) & [Article 32](https://gdpr-info.eu/art-32-gdpr/) provides requirements for adequate security, and minimizing the data collected and processed.
 - [Article 30](https://gdpr-info.eu/art-30-gdpr/) provides requirements for controllers and processors to keep a written record of their data processing and security policies, to be given to the auhorities on request. This does not apply to organizations under 250 employees that only do "occasional" processing, are not likely to result in a risk to rights and freedoms and does not process special personal data. This requirement applies to us, since we'll be processing medical data (at least food allergies). The required record is similar to the information to be supplied to data subjects, so this requirement can probably be fulfilled using the privacy policy.
 - [Article 33](https://gdpr-info.eu/art-34-gdpr/) and [article 34](https://gdpr-info.eu/art-34-gdpr/) provide rules on communicating data breaches to the authorities and the data subjects.
 - [Article 35](https://gdpr-info.eu/art-35-gdpr/) requires a "data protection impact assessment" (DPIA) when data processing poses a high risk to rights and freedoms. In particular, this is required when processing special personal data on a large scale. I would say this does not apply to our handling of (medical) data.
 - [Article 37](https://gdpr-info.eu/art-37-gdpr/)/38/39 have rules for designating a data protection officer (DPO) in some cases. None of these would apply to us.
 - Chapter 5 provides rules for transferring persona data outside of the EU. We have no plans to do so.
 - Chapter 6 & 7 provide rules about supervisory authorities.
 - Chapter 8 provides rules about resolution of conflicts and complaints.
 - Chapter 9 provides rules for specific situations, none of which apply to us.
 - Chapter 10 and 11 provide some legalese that does not seem to be relevant to us.

**Comment by matthijskooijman** [comment]
Here is some more info on legitimate interest (6.1f): https://ico.org.uk/for-organisations/guide-to-the-general-data-protection-regulation-gdpr/legitimate-interests/when-can-we-rely-on-legitimate-interests/ (which I haven't really read yet).

**Comment by matthijskooijman** [comment]
Note that, in addition to the GDPR, the PECR rules for e-mail marketing also apply: https://ico.org.uk/for-organisations/guide-to-pecr/electronic-and-telephone-marketing/electronic-mail-marketing/

In short, they say you can only send people e-mail when you have explicit consent (opt-in) or if you have some contract with them (sometimes called "soft opt-in"). In both cases, you must offer opt-out. For our case, this means that anyone who has visited one of our events falls under soft opt-in. Since we used registration in our system as a way to sign up for event announcements, we are assuming that anyone who is registered in our system has opted-in to these announcements (which does *not* comply with the GDPR-definition of consent, and probably not under previous laws either).

For the future, we should really get explicit consent for sending event announcements. I suggest we do this for all new users, as well as returning users once they log in.

For existing users that attended an event, we can still lawfully send them e-mail based on the soft opt-in. For existing users that did not attend an event, we can technically not send them any marketing e-mail. We could consider ignoring that for these users, or send them a one-time e-mail to ask for their explicit consent (which might be ok if it is not a "marketing email", not sure what the exact scope of PECR is). It might be good to send this one-off e-mail to all users, to ask all of them to make their consent explicit. This could be combined with the announcement of our new system, and pushing GPDR-mandate privacy notices to our users.

**Comment by matthijskooijman** [comment]
On additional thing to think about is storing emergency contact details. Given that this is data about third parties, the user that enters them is not in a position to give consent about storing these details. Instead, the more likely legal basis for storing this data is the legitimate interest (under 6.1f). A similar example is given at https://ico.org.uk/for-organisations/guide-to-the-general-data-protection-regulation-gdpr/legitimate-interests/how-do-we-apply-legitimate-interests-in-practice/ (down at "3. How do we do the balancing test?").

In this case, I'm not sure how the requirements about informing the data subjects would fall out. It might be possible to argue it is not feasible or not reasonable to inform them about storing their data.

**Comment by matthijskooijman** [comment]
Here's some good info on how and what to communicate about our usage of data: https://ico.org.uk/for-organisations/guide-to-the-general-data-protection-regulation-gdpr/individual-rights/right-to-be-informed/

**Comment by matthijskooijman** [comment]
For consent, we must also keep a clear record when consent is given. This should at least contain who, when and how they gave consent, including all relevant information given at the time (the exact wording of the checkbox, any versioned references to the privacy policy, etc.). This suggests that keeping a privacy policy in git (using hashes to unamiguously refer to them) might be useful (though linking to github for a reference might not be ideal).

Also, this implies that when updates happen to the consent form or related privacy documents, the consent given changes. This means we will not only need to log an audit trail of the exact consent given, but probably also include a version number for the consent in the database itself, so we can actively filter on that (e.g. only do certain processing for people who gave the latest version of the consent).

**Comment by matthijskooijman** [comment]
This is probably a matter of adding a model with date, user, consent name, consent description (i.e. what was behind the checkbox) and consented/withdrawn.

This should track consent for e-mail announcements.

Consent for terms & conditions when registering for an event might not be strictly required, but might be just as easy, so might as well include that as well (maybe the consent model should include an optional reference to Registration then?

**Comment by matthijskooijman** [comment]
Next step here is probably to review the list of things in the first post to see if our current plans and implementation are compliant.

**Comment by matthijskooijman** [comment]
I looked through the first post, here's some things that are still relevant.

We collect medical data on basis of explicit consent, so this consent must be explicitly given using a checkbox or similar. Filling in data in an optional field is consent, which is sufficient for normal data, but [probably not *explicit* consent](https://ico.org.uk/for-organisations/guide-to-the-general-data-protection-regulation-gdpr/consent/what-is-valid-consent/#what5) needed for medical data. I created #99 for this.


> [Article 21](https://gdpr-info.eu/art-21-gdpr/) specifies the right to object to processing when data is collected for "legitimate purposes" (6.1f) or public interest (6.1e), or when data is used for direct marketing. This might apply to us, in which case we must at least point out this possibility to our users.

This would apply to:
 - Sending event announcements, if we view that as "direct marketing" (have not quickly found a good definition of that). According to the letter we should notify users that they have the right to object, but since we are processing according to consent, we will be notify users that they can withdraw consent, which I think removes the need for using the right to object entirely.
 - The emergency contacts, which are used on basis of legitimate interests. The right to object must be made known to the data subject (which in this case is the *emergency contact*, not the player) at earliest communication. IOW when calling someone in a case of emergency we should notify them of their right to object, which does not seem reasonable to me, so I guess we can can just ignore this now (maybe put a notice in our privacy statement, but even that does not seem useful).

> [Article 30](https://gdpr-info.eu/art-30-gdpr/) provides requirements for controllers and processors to keep a written record of their data processing and security policies, to be given to the auhorities on request. This does not apply to organizations under 250 employees that only do "occasional" processing, are not likely to result in a risk to rights and freedoms and does not process special personal data. This requirement applies to us, since we'll be processing medical data (at least food allergies). The required record is similar to the information to be supplied to data subjects, so this requirement can probably be fulfilled using the privacy policy.

We might want to revisit this one later. I think some of the considerations in this issue would be good to write down somewhere, but they should probably not further complicate the privacy statement. Alternatively, our provicy statement could be somwhat simplified into a TL;DR version, with the full GDPR-specific details under the fold.


With the above, I think this issue is sufficiently covered for the MVP, so I'm removing it from the MVP project.
